File: audience/src/main/java/br/com/concretesolutions/audience/core/concurrent/ActorThread.java
package br.com.concretesolutions.audience.core.concurrent;

import android.os.HandlerThread;

public final class ActorThread extends HandlerThread {

    private BaseHandler messageHandler;

    public ActorThread() {
        super("ActorStorage-Thread");
    }

    @Override
    public synchronized void start() {
        super.start();
        messageHandler = new BGInboxHandler(getLooper());
    }

    public BaseHandler getMessageHandler() {
        return messageHandler;
    }
}

File: audience-core/src/main/kotlin/io/github/victorolinasc/audience/core/concurrent/ActorThread.kt
package io.github.victorolinasc.audience.core.concurrent

import android.os.HandlerThread

class ActorThread : HandlerThread("ActorStorage-Thread") {

    var messageHandler: BaseHandler? = null
        private set

    @Synchronized
    override fun start() {
        super.start()
        messageHandler = BGInboxHandler(looper)
    }
}